File: src/main/webapp/resources/assets/js/app/services/salonService.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
var servicesModule = angular.module('saircoAppServices');

servicesModule.factory('salonService', ['$http', function ($http) {
        
        return {
            apiUrl: apiUrl,
            getAllSalones: function () {
                return $http.get(this.apiUrl+'salon/');
            },
            getByEquipoId: function (EquipoBySalonId) {
                return $http.get(this.apiUrl + 'equipo/' + equipoId);
            },
            getSalonById: function (salonId) {
                return $http.get(this.apiUrl+'salon/'+salonId);
            },
            createSalon: function (salon) {
                return $http.put(this.apiUrl+'salon/', salon);
            }
        };
    }]);

File: src/main/java/co/utb/softeng/moviesapp/services/SalonService.java

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package co.utb.softeng.moviesapp.services;

import co.utb.softeng.moviesapp.entities.Equipo;
import co.utb.softeng.moviesapp.entities.Salon;
import java.util.List;

/**
 *
 * @author William
 */
public interface SalonService {
    
    public List<Salon> getAllSalones();
    public Salon getSalonById(Long id);
    public List<Salon> getSalonByName(String name);
    public List<Salon> getSalonByEquipoId(Long equipoId);
    public List<Salon> getSalonByEquipoName(String equipoName);
    public void saveOrUpdateSalon(Salon salon);
    
    public Salon addEquiposToSalon(List<Equipo> equipos, Long salonId);
    public void deleteSalon(Long id);
    
    
}